File: src/main/java/io/pageobject/generator/locator/NgForLocatorPartGenerator.java
package io.pageobject.generator.locator;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.locator.protractor.CssLocatorSource;
import io.pageobject.generator.locator.protractor.NameLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.locator.LocatorPart.prefixPart;

public class NgForLocatorPartGenerator extends RepeaterLocatorPartGenerator {

    private LocatorPart locatorPart;

    public NgForLocatorPartGenerator(Element element) {
        super(element, new NameLocatorSource(), new CssLocatorSource());
    }

    @Override
    public LocatorPart firstPart(GeneratorContext context) {
        locatorPart = super.firstPart(context);
        return locatorPart;
    }

    @Override
    public LocatorPart lastPart(GeneratorContext context, int index) {
        locatorPart = super.lastPart(context, index);
        return locatorPart;
    }

    @Override
    public LocatorPart middlePart(GeneratorContext context, int index) {
        StringBuilder value = new StringBuilder(locatorPart.getPart());
        value.append(".get(rowIndex");
        value.append(index + 1);
        value.append(")");
        return prefixPart(value.toString());
    }
}

File: lib/locator/ng-for-locator-part-generator.js
"use strict";

const RepeaterLocatorPartGenerator = require('./repeater-locator-part-generator');
const NameLocatorSource = require('./protractor/name-locator-source');
const CssLocatorSource = require('./protractor/css-locator-source');
const prefixPart = require('./locator-part').prefixPart;

class NgForLocatorPartGenerator extends RepeaterLocatorPartGenerator {

  constructor(element) {
    super(element, new NameLocatorSource(), new CssLocatorSource());
  }

  firstPart(context) {
    this._locatorPart = super.firstPart(context);
    return this._locatorPart;
  }

  lastPart(context, index) {
    this._locatorPart = super.lastPart(context, index);
    return this._locatorPart;
  }

  middlePart(context, index) {
    return prefixPart(`${this._locatorPart.part}.get(rowIndex${index + 1})`);
  }

}

module.exports = NgForLocatorPartGenerator;